### PR TITLE
chore(task-processor): add better logging for failed tasks

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Adds better logging for failed tasks to help investigations where needed. 

## How did you test this code?

Updated an existing test to check for the correct log messages.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ services:
       start_period: 20s
 
   flagsmith:
```
Comment (gagantrivedi) [inline docker-compose.yml]: We need to revert this, right?

Comment (matthewelwell) [reply docker-compose.yml]: Yes, good catch - thank you.

Comment (matthewelwell) [reply docker-compose.yml:25]: Yes, good catch - thank you.

Comment (matthewelwell) [reply docker-compose.yml:25]: Yes, good catch - thank you.

**Review by gagantrivedi — APPROVED** [review]
Can merge after docker-compose changes

File: docker-compose.yml (hunk 1 above)
Comment (matthewelwell) [reply docker-compose.yml]: Done

**Pull request merged**